### PR TITLE
Add larger test for fillOverlapHamiltonianMatrices

### PR DESCRIPTION
**Pull request by markdewing**
The data for this test is from 10 samples and 12 variational parameters.
The data is stored as text ("diamond_fill_data.h") to mesh better with the git repository.  It was converted from HDF to text with the convert_hdf_to_cpp.py script.

This PR replaces the test additions in #3352. 

## What type(s) of changes does this code introduce?
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by markdewing** [comment]
test this please

File: src/QMCDrivers/tests/diamond_fill_data.h [hunk 1]
```diff
@@ -0,0 +1,171 @@
+
+// Generated with convert_hdf_to_cpp.py
+// clang-format off
+
+void get_diamond_fill_data(FillData &fd)
```
Comment (ye-luo) [inline src/QMCDrivers/tests/diamond_fill_data.h:5]: Need basic documentation about what this function do.

File: src/QMCDrivers/tests/test_QMCCostFunctionBatched.cpp [hunk 2]
```diff
@@ -126,5 +131,89 @@ TEST_CASE("fillOverlapAndHamiltonianMatrices", "[drivers]")
...
+  std::vector<double> energy_new;
+
+  Matrix<double> derivRecords;
+  Matrix<double> HDerivRecords;
+
+  Matrix<double> ovlp_gold;
+  Matrix<double> ham_gold;
+};
+
+void fill_from_text(int num_opt_crowds, FillData& fd)
```
Comment (ye-luo) [inline src/QMCDrivers/tests/test_QMCCostFunctionBatched.cpp:152]: Need basic documentation about what this function do.

File: src/QMCDrivers/tests/test_QMCCostFunctionBatched.cpp [hunk 3]
```diff
@@ -126,5 +131,89 @@ TEST_CASE("fillOverlapAndHamiltonianMatrices", "[drivers]")
...
+    {
+      //app_log() << "iw = " << iw << " iw2 = " << iw2 << " ovlp = " << ovlp(iw,iw2) << " " << ovlp_gold(iw,iw2);
+      //app_log() << " ham = " << ham(iw,iw2) << " " << ham_gold(iw,iw2) << std::endl;
+      CHECK(ovlp(iw, iw2) == Approx(fd.ovlp_gold(iw, iw2)));
+      CHECK(ham(iw, iw2) == Approx(fd.ham_gold(iw, iw2)));
+    }
+  }
+}
+
+#include "diamond_fill_data.h"
```
Comment (ye-luo) [inline src/QMCDrivers/tests/test_QMCCostFunctionBatched.cpp:201]: Can we move this to the top.

**Review by prckent — APPROVED** [review]
Looks to me that the requests have all been addressed. Thanks Mark!

**Comment by prckent** [comment]
Start testing in-house

**Pull request merged**